src/libs/fsdrv/lv_fs_littlefs.c

```diff
@@ -176,13 +177,98 @@ static lv_fs_res_t fs_seek(lv_fs_drv_t * drv, void * file_p, uint32_t pos, lv_fs
  */
 static lv_fs_res_t fs_tell(lv_fs_drv_t * drv, void * file_p, uint32_t * pos_p)
 {
-    LV_UNUSED(drv);
-    LittleFile * lf = (LittleFile *)file_p;
+    LittleFile * lf = file_p;
 
-    lfs_t * lfs = (lfs_t *)drv->user_data;
+    lfs_t * lfs = drv->user_data;
     *pos_p = lfs_file_tell(lfs, &lf->file);
 
     return (int32_t)(*pos_p) < 0 ? LV_FS_RES_UNKNOWN : LV_FS_RES_OK;
 }
 
+/**
+ * Open a directory
+ * @param drv       pointer to a driver where this function belongs
+ * @param path      path to the directory beginning with the driver letter (e.g. S:/folder)
+ * @return          a directory descriptor or NULL on error
+ */
+static void * fs_dir_open(lv_fs_drv_t * drv, const char * path)
+{
+    LittleDirectory * ld = lv_malloc(sizeof(LittleDirectory));
+    LV_ASSERT_MALLOC(ld);
+
+    lfs_t * lfs = drv->user_data;
+    int err = lfs_dir_open(lfs, &ld->dir, path);
+    if(err != LFS_ERR_OK) {
+        lv_free(ld);
+        return NULL;
+    }
+
+    return (void *)ld;
+}
+
+/**
+ * Close an opened directory
+ * @param drv      pointer to a driver where this function belongs
+ * @param dir_p    pointer to a dir_p variable. (opened with fs_dir_open)
+ * @return         LV_FS_RES_OK: no error or any error from @lv_fs_res_t enum
+ */
+static lv_fs_res_t fs_dir_close(lv_fs_drv_t * drv, void * dir_p)
+{
+    LittleDirectory * ld = dir_p;
+
+    lfs_t * lfs = drv->user_data;
+    int rc = lfs_dir_close(lfs, &ld->dir);
+
+    if(rc < 0) return LV_FS_RES_UNKNOWN;
+    lv_free(ld);
+
+    return LV_FS_RES_OK;
+}
+
+/**
+ * Read data from an opened directory
+ * @param drv      pointer to a driver where this function belongs
+ * @param dir_p    pointer to a file_t variable.
+ * @param fn       pointer to a buffer to store the filename
+ * @param fn_len   length of the buffer to store the filename
+ * @return         LV_FS_RES_OK: no error or any error from @lv_fs_res_t enum
+ */
+static lv_fs_res_t fs_dir_read(lv_fs_drv_t * drv, void * dir_p, char * fn, uint32_t fn_len)
+{
+    if(fn_len == 0) return LV_FS_RES_INV_PARAM;
+
+    LittleDirectory * lf = dir_p;
+
+    lfs_t * lfs = drv->user_data;
+
+    fn[0] = '\0';
+    do {
+        struct lfs_info info;
+        int res = lfs_dir_read(lfs, &lf->dir, &info);
+
+        if(res < 0) return LV_FS_RES_UNKNOWN;
+        if(res == 0) { /* End of the directory */
+            fn[0] = '\0';
+            break;
+        }
+
+        if(info.type != LFS_TYPE_DIR) {
+            lv_strncpy(fn, info.name, fn_len - 1);
+            fn[fn_len - 1] = '\0';
```

Comment:
Since you rebased, let's use `lv_strlcpy`. Sorry for the hassle.
```suggestion
            lv_strlcpy(fn, info.name, fn_len);
```

Reply:
Done, no worries, everything you say helps me to better understand how C/lvgl work :pray: